### DISCUSSION TITLE
Can't expand nested fields when querying product variant Store API route

### DISCUSSION THREAD

**Discussion by AlexTechNoir** [medusajs/medusa#5963] [Q&A]
Good day!

I'm querying Store API route of product variant, and I need to expand `options.option` nested field (to get the name of the option itself). Docs [say](https://docs.medusajs.com/api/store#product-variants_getvariants) that, in the response, `options` is expandable field, as well as `option` inside of it

When I query it like this: `http://localhost:9000/store/variants?expand=options.option` it says "Relations [options.option] are not valid". Previously, I successfully expanded nested fields, like `variants.prices` in Products API, but here it doesn't work.

What am I doing wrong?

**Comment by hauseyo** [comment]
Hi,
Facing the same issue. Is there any solutions for it?

**Comment by hauseyo** [comment]
What I found is that it's impossible to expand nested fields querying product variant, I guess
In order to get option, that array should contain options and options.option. But I may be wrong

https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/api/routes/store/variants/index.ts#L42C1-L42C77

**Comment by AlexTechNoir** [comment]
@hauseyo Hello! Sorry, I haven't found anything to solve this

I wanted to implement a **list of product variant items**, instead of a **list of product items** that include variants, but after I've learned that product variants are not searchable, unlike products, I gave up on this

Based on what you've found, fields are indeed not expandable

So, if you need to get option names/titles, the only way is through product items list, probably (`/store/products?expand=options` -> `products[i].options`)
